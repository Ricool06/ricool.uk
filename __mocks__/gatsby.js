import React from 'react';

const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(queryString => queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) => React.createElement('a', {
    ...rest,
    href: to,
  })),
  StaticQuery: jest.fn(),
};
