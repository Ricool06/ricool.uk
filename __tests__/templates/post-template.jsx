import React from 'react';
import { render, cleanup } from 'react-testing-library';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import PurePostTemplate from '../../src/templates/post-template';

describe('PostTemplate', () => {
  const staticQueryData = {
    site: {
      siteMetadata: {
        title: 'Ricool\'s Blog',
        description: 'making things all day',
        author: 'Ricool06',
      },
    },
  };

  afterEach(cleanup);

  it('should have correct header text, set by title', () => {
    StaticQuery.mockImplementationOnce(({ render: mockRender }) => mockRender(staticQueryData));

    const data = {
      markdownRemark: {
        frontmatter: {
          title: 'My First Blog Post',
          date: '2019-26-01',
          html: '<p>This is the blog content</p>',
        },
      },
    };
    const { container } = render(<PurePostTemplate data={data} />);

    const headers = container.querySelectorAll('h1');
    const date = container.querySelector('h2');

    expect(headers[0].textContent).toBe(staticQueryData.site.siteMetadata.title);
    expect(headers[1].textContent).toBe(data.markdownRemark.frontmatter.title);
    expect(date.textContent).toBe(data.markdownRemark.frontmatter.date);
  });

  it('should have a correctly configured StaticQuery component for default export', () => {
    StaticQuery.mockImplementationOnce(({ render: mockRender }) => mockRender(staticQueryData));

    const data = {
      markdownRemark: {
        frontmatter: {
          title: 'My First Blog Post',
          date: '2019-26-01',
          html: '<p>This is the blog content</p>',
        },
      },
    };
    const jsonSnapshot = renderer.create(<PurePostTemplate data={data} />).toJSON();

    expect(jsonSnapshot).toMatchSnapshot();
  });
});
