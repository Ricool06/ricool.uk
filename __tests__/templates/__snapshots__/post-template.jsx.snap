// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate should have a correctly configured StaticQuery component for default export 1`] = `
Array [
  <div
    style={
      Object {
        "background": "#339988",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Ricool's Blog
        </a>
      </h1>
    </div>
  </div>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <div
      className="blog-post-container"
    >
      <div
        className="blog-post"
      >
        <h1>
          My First Blog Post
        </h1>
        <h2>
          2019-26-01
        </h2>
        <div
          className="blog-post-content"
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
      </div>
    </div>
    <footer>
      Â©
      2019
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
